iceoryx_meta/CMakeLists.txt

```diff
@@ -43,6 +44,18 @@ if(coverage)
     endif(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
 endif(coverage)
 
+if(sanitize)
```

Comment:
this won't work if iceoryx is not build with `iceoryx_meta`. Maybe it should be moved to `iceoryx_utils/cmake/IceoryxPlatformDetection.cmake` ... and maybe we need to rename that file to make it clear that there is more than just platform detection in in

Reply:
`sanitize` is a special build (invoked in github action of its own). What would be the need to support it for other builds ?

Btw what are build you are referring to @elBoberido (colcon build?)

Reply:
if you build just iceoryx_utils for example, then the `${SANITIZE_COMPILE_FLAGS}` variable would be empty. If it's put into `IceoryxPlatformDetection.cmake` it would still work

Reply:
Makes sense 
In that case other compiler settings  (for `coverage` ) can also be moved to `IceoryxPlatformDetection.cmake`.

**Whats a good name for this file ?**
`IceoryxCentral.cmake`
Any other suggestions ?

Reply:
@elBoberido i think it make more sense to leave it here, because iceoryx_meta is some kind of "main" cmake file to build iceoryx with. Placing that into iceoryx_utils/cmake/IceoryxPlatformDetection.cmake will "hide" this option. 
For me it is ok to have it here, because most of the users currently use iceoryx_meta for building. 
Discussion is related to https://github.com/eclipse/iceoryx/issues/230.

Reply:
The `coverage` is just used in the tests, right? As far as I know, this also needs some configuration passed to cmake.

To be able to use the sanitizer might be useful outside of the tests and iceoryx_meta, e.g. when someone integrates iceoryx into a custom framework and wants to switch the sanitize on. Therefore I would move it some layers down. We can still have the option in iceoryx_meta.

How about calling the file just `IceoryxPlatform.cmake`?

Reply:
@elBoberido `coverage` flags are used for compiling the sources (may be dependent libs are linked against tests only)
For now `IceoryxPlatformDetection.cmake` contains platform specific compiler flags. `sanitize` flags are compiler dependent not platform.

@dkroenke @elBoberido do I move `sanitize` flags ?

Reply:
I think we have a tie here. Where is the tiebreaker 😄

Reply:
> For now `IceoryxPlatformDetection.cmake` contains platform specific compiler flags. `sanitize` flags are compiler dependent not platform.

Well, depending on the platform different flags must be set, so it wouldn't be that unrelated.